### PR TITLE
BAN-2160 Show links in captions

### PR DESCRIPTION
**Pull request by damiendillon**
Add support for links in captions of images and embed features.

https://mydevex.atlassian.net/browse/BAN-2160

### CODE REVIEW

**Comment by dsager** [comment]
LGTM.
The code for captions looks pretty similar, I wonder if it makes sense to share it between `Embed` and `Image`?

File: spec/article_json/export/apple_news/elements/image_spec.rb [hunk 1]
```diff
@@ -33,17 +35,29 @@
...
+        end
+
         it { should eq expected_json }
       end
 
-      context 'and when the image is floating on the right' do
-        let(:float) { :right }
+      context 'and the caption has got a link' do
+        let(:caption_text) do
+          "This has a link: <a href=\"http://en.wikipedia.org/wiki/Lorem_ipsum\">Lorem ipsum</a>"
```
Comment (ozgun) [inline spec/article_json/export/apple_news/elements/image_spec.rb:59]: I'd use single quotes:
```ruby
'This has a link: <a href="http://en.wikipedia.org/wiki/Lorem_ipsum">Lorem ipsum</a>'
```

File: lib/article_json/export/apple_news/elements/image.rb [hunk 2]
```diff
@@ -26,15 +26,30 @@ def caption
...
           # @return [String]
           def caption_text
-            @element.caption.first&.content
+            text.empty? ? nil : text
+          end
+
+          # @return [String]
+          def text
+            @element.caption.map do |child_element|
+              text_exporter.new(child_element)
```
Comment (ozgun) [inline lib/article_json/export/apple_news/elements/image.rb:50]: This looks like a one liner to me :)

**Comment by ozgun** [comment]
+1 Looks good, I've just added a few minor comments, thanks!

**Comment by ozgun** [comment]
> The code for captions looks pretty similar, I wonder if it makes sense to share it between Embed and Image?

Good idea, but if it's gonna take time, I'd say let's merge this now and refactor later because there is already a ticket waiting for this fix.

**Comment by damiendillon** [comment]
> LGTM.
> The code for captions looks pretty similar, I wonder if it makes sense to share it between `Embed` and `Image`?

Definitely does. I am pretty confident I have almost the same code in one other place too, but like Oz said, we wanted it to quickly as it is a reopened ticket and this was not really in the original scope as I had not factored links so we want to close it. 

I will give all the article json stuff a refactoring/ cleanup with Rubocop soon.

**Pull request merged**